snow/engine/avalanche/transitive_test.go

```diff
@@ -4077,3 +4077,264 @@ func TestEngineDoubleChit(t *testing.T) {
 		t.Fatalf("Wrong tx status: %s ; expected: %s", status, choices.Accepted)
 	}
 }
+
+func TestEngineInvalidChit(t *testing.T) {
+	config := DefaultConfig()
+	config.Params.BetaVirtuous = 2
+	config.Params.BetaRogue = 2
+	config.Params.Alpha = 2
+	config.Params.K = 2
+	vals := validators.NewSet()
+	config.Validators = vals
+	vdr0 := ids.GenerateTestShortID()
+	vdr1 := ids.GenerateTestShortID()
+	if err := vals.AddWeight(vdr0, 1); err != nil {
+		t.Fatal(err)
+	}
+	if err := vals.AddWeight(vdr1, 1); err != nil {
+		t.Fatal(err)
+	}
+	sender := &common.SenderTest{}
+	sender.T = t
+	config.Sender = sender
+	sender.Default(true)
+	sender.CantGetAcceptedFrontier = false
+	manager := vertex.NewTestManager(t)
+	config.Manager = manager
+	manager.Default(true)
+	gVtx := &avalanche.TestVertex{TestDecidable: choices.TestDecidable{
+		IDV:     ids.GenerateTestID(),
+		StatusV: choices.Accepted,
+	}}
+	mVtx := &avalanche.TestVertex{TestDecidable: choices.TestDecidable{
+		IDV:     ids.GenerateTestID(),
+		StatusV: choices.Accepted,
+	}}
+	manager.EdgeF = func() []ids.ID { return []ids.ID{gVtx.ID(), mVtx.ID()} }
+	manager.GetF = func(id ids.ID) (avalanche.Vertex, error) {
+		switch id {
+		case gVtx.ID():
+			return gVtx, nil
+		case mVtx.ID():
+			return mVtx, nil
+		}
+		t.Fatalf("Unknown vertex")
```

Comment:
In the new tests I write I've been using the `assert` library because the interface is a little nicer.
Here, for example, I would do:
```
assert.FailNow(t, "Unknown vertex")
```

nabd if you want to keep this test as is since we're already using `t.Fatalf` in most tests